### PR TITLE
Mobile Header Base

### PR DESCRIPTION
**Pull request by ModulesUnraveled**
### Description of work
- [YALB-86: Utility Nav](https://yaleits.atlassian.net/browse/YALB-86) is temporarily blocked by some design discussions, so this PR abstracts some of the "shared" mobile header and menu stuff so that I can work on breadcrumbs and the main menu without being blocked by the utility nav. So, this includes:
- Menu base templates (not demoed anywhere, since they'll be used by each menu, but don't do anything by themselves)
- Mobile Menu toggle
- Updates to the site-header to support mobile designs

### Testing Link(s)
- [ ] Navigate to the [Full Page Layout](https://deploy-preview-30--dev-component-library-twig.netlify.app/?path=/story/page-layouts-page-layouts--full-width) preview

### Functional Review Steps
- [ ] Verify everything works as expected on small screens

### Design Review
- [ ] Verify the designs match the [Figma Designs](https://www.figma.com/file/OVfmCW5s1gI7xnrM8ibX2y/UI-Kit-Alpha-%5BYaleSites%5D)

### Accessibility Review
- [ ] Verify the component meets Accessibility requirements

### CODE REVIEW

**Review by kara-franco — CHANGES_REQUESTED** [review]
Mobile menu toggle: The ARIA states are not being announced to screen readers (expanded, collapsed, pressed etc). 

Keyboard controls are great though! I can open and close the toggle with space, enter; and close with esc.

**Comment by miketullo95** [comment]
@ahitson what is your opinion on having the primary navigation centered on mobile even when you have right or left aligned selected. It might be hard for a user to tap right right/left-most item if its that close to the screen edge. 

image

**Comment by ModulesUnraveled** [comment]
@miketullo95 the primary nav will look very different from this on mobile. What's there now is just a placeholder block. I don't think you'll have a concern once it's actually built  (hopefully next week)

**Comment by ModulesUnraveled** [comment]
@kara-franco I think I've set the button aria attributes appropriately.

**Review by ahitson — APPROVED** [review]

File: components/01-atoms/buttons/_buttons.scss [hunk 1]
```diff
@@ -8,6 +10,15 @@ $button-outline-weights: (
   '4': var(--border-thickness-4),
 );
 
+// The button-reset mixin is used to remove browser default styles for button
+// elements that shouldn't look like a traditional "button".
+@mixin button-reset {
```
Comment (callinmullaney) [inline components/01-atoms/buttons/_buttons.scss:15]: Similar to `list-reset` comment I had earlier. Wouldn't this be namespaced during implementation to `button.reset`? On my local I was able to:
1. Change this to `@include reset  { ....` 
2. Add `@use '../../../01-atoms/buttons/buttons' as button;` at the top of the `_menu-toggle.scss` file
3. Swap out the mixin reference to `@include button.reset;`

File: components/01-atoms/buttons/button.twig [hunk 2]
```diff
@@ -40,8 +41,15 @@
   }) %}
 {% endif %}
 
-<button {{ add_attributes(button__attributes) }}>
+{% if button__href %}
```
Comment (callinmullaney) [inline components/01-atoms/buttons/button.twig:44]: This looks great! Easy to use a `<a>` element here when needed.

File: components/01-atoms/buttons/_buttons.scss (hunk 1 above)
Comment (ModulesUnraveled) [reply components/01-atoms/buttons/_buttons.scss:15]: I did this in the Breadcrumbs PR when I refactored all of the `@import`s to `@use`s. https://github.com/yalesites-org/component-library-twig/pull/31/files#diff-e6b4386e20a680dd5b7e303065b8387277192ebb6940cc14cfe6d3a3b1e1f113

**Review by callinmullaney — APPROVED** [review]
@ModulesUnraveled looks good. Approved.

**Comment by kara-franco** [comment]
@ModulesUnraveled 
The expanded state continues to not be announced by a screen reader. My hunch is that the aria-pressed state is overriding the expanded announcement (and upon further reading, aria-pressed should only be used for true toggle buttons). 

**I suggest removing the aria-pressed attribute, as well as the aria-controls** (I did more reading and this ARIA attribute has poor support and may not be interpreted the same across assistive tech). 

Further reading: 

- [https://a11ysupport.io/tests/tech__aria__aria-controls](https://a11ysupport.io/tests/tech__aria__aria-controls)
- [https://heydonworks.com/article/aria-controls-is-poop/](https://heydonworks.com/article/aria-controls-is-poop/) (as of JAWS 2018, exposure of aria-controls has been removed). 
- [https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/pressed/#do-not-use-for-marking-expandability](https://www.accessibility-developer-guide.com/examples/sensible-aria-usage/pressed/#do-not-use-for-marking-expandability)
- [https://adrianroselli.com/2019/08/under-engineered-toggles-too.html](https://adrianroselli.com/2019/08/under-engineered-toggles-too.html) (nice write up of how to use aria-pressed for toggle buttons)

**Comment by ModulesUnraveled** [comment]
@kara-franco I've removed those aria attributes. Let me know if it's working as expected now.

Also, can you elaborate (here or in Teams) on what you mean by "aria-pressed should only be used for true toggle buttons"

**Comment by kara-franco** [comment]
@ModulesUnraveled 
Awesome, the menu button is announcing expanded/collapsed now in JAWS and VO. 

As for a true toggle button: I mean a button that switches between two states, off and on vs a menu button. The menu button is following a disclosure pattern, where content is hidden and then exposed from the interaction. At first glance, they seem the same, but the outcome of the interaction is announced differently to blind people. 

- toggle button with dark mode or light mode using aria-pressed: announced as selected 
- hamburger menu button that opens a list of nav links using aria-expanded: announced as expanded, announced as collapsed once closed.

**Review by kara-franco — APPROVED** [review]

**Pull request merged**